solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -972,7 +978,8 @@ private DocSet getAndCacheDocSet(Query query) throws IOException {
     }
 
     DocSet answer;
-    if (SolrQueryTimeoutImpl.getInstance().isTimeoutEnabled()) {
+    SolrRequestInfo requestInfo = SolrRequestInfo.getRequestInfo();
+    if (requestInfo != null && requestInfo.getLimits().isTimeoutEnabled()) {
```

Comment:
The previous pattern had the nice characteristic that it's a one-liner.  We could put the use of SolrRequestInfo on a limits class convenience method?  Not a big deal anyway.

Reply:
in some cases there's an assert for requestInfo != null, rather than a null check but that assert is not appropriate in all cases.